New data model

6iruk opened the pull request.

bereketnyb commented on api/views.py:
```diff
@@ -12,188 +12,175 @@
+                        announcements = announcements | section.announcement_set.filter(query)
+
+        # Apply filter
+        if request.GET.get('id'):
```
We need the distinct filter here. I don't know how, but there will be duplicate announcements.

6iruk replied:
okay, now when i think about it there could be a possibility of duplicates.

addismiraph committed:
create new base

remove unused file: index_temp.html

create new base html to replace old one

addismiraph committed:
write boilerplate

the bare bones is done

addismiraph committed:
custom css

create file to write the css

addismiraph committed:
add navbar

add bootstrap navbar and include push.css

addismiraph committed:
new base

rename the old base html to abase and make the new one live. also start writing the custom css push.css

addismiraph committed:
add fonts for design

proxima nova is the brand fonts - bold, semibold and regular are required.

addismiraph committed:
remove other fonts

addismiraph committed:
add fonts to css

rename fonts and add @font-face rules

addismiraph committed:
set font size and colors

define classes for use

addismiraph committed:
add bottom links

addismiraph committed:
home page link

add home page link to logo and section code block

addismiraph committed:
block display

make display block for all the writing classes except meta class

addismiraph committed:
new section code

addismiraph committed:
remove padding

remove padding and margins added by bootstrap!

addismiraph committed:
make footer

make the footer it's own row

addismiraph committed:
announcement block

make announcement a full block on its own and have 'latest uploads' and 'courses' share a row under it. all of them stack in mobile view

addismiraph committed:
sass mixins

download sass mixins to ensure cross browser compatiblity

addismiraph committed:
course list snippet

separate the course list menu into an html snippet for reusability

addismiraph committed:
new course view

code the new design and replace the old one

addismiraph committed:
main sass file

create variables for push brand colors. use mixins to make cross browser compatibility

addismiraph committed:
space under header

add a spacer div that will push everything in the content block lower. so the header will look nice. gives it a good margin

addismiraph committed:
adding spacers

remove margin property from h1 tag and use spacer div instead

addismiraph committed:
rendered css

addismiraph committed:
multiple color gradient mixin

the current mixin only takes two colors but we need four, so modify it.

addismiraph committed:
remove old base html

addismiraph committed:
multiple download page

create the new page dedicated to multiple downloads. a new view needs to be written to handle this.

addismiraph committed:
smaller font size

correct the font size to be just right

addismiraph committed:
change 'section code' to 'extra'

change section code block to extra so that it can be used in other places. so the base is more reusable.

addismiraph committed:
move script tag to bottom

and add href addresses for links are the bottom

addismiraph committed:
use more friendly time system

use Django's humanize application to make timestamps more beautiful.

addismiraph committed:
new index

this home page is only temporary.

addismiraph committed:
add humanize app

addismiraph committed:
add stripe to table

makes it readable

addismiraph committed:
remove unneeded files

addismiraph committed:
change grouping from departement to section

Fetching and displaying the courses grouped by each section that the teacher is teaching.

Co-Authored-By: Biruk Tamiru <6iruk@users.noreply.github.com>

6iruk committed:
Change prev_dept to prev_section on line 112

Missed this last time

6iruk committed:
Implement new data model

-New data model has been implemented
-Views.py has been modified to accommodate new data model
-The api has also been changed
-Templates for the student's side have been updated

6iruk committed:
Update teachers side

6iruk committed:
Configure MySql

6iruk committed:
configure settings
